### PR TITLE
 improve github actions build 

### PR DESCRIPTION
**Pull request by dgsiegel**
this enables buildroot to get current (annotated) tags, #commits since then and current hash and add that information to /etc/os-release

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -16,6 +16,8 @@ jobs:
     env:
       BUILDROOT_VERSION: '2020.08.2'
       BUILDROOT_DIR: 'buildroot'
+      # Stupid hacke to invalidate cache from UI
```
Comment (htruong) [inline .github/workflows/release.yml:19]: nit: hack

**Comment by htruong** [comment]
Bigger question remains as is there any way that we can force the build to build everything from a fresh cache?

**Comment by htruong** [comment]
How do we inject the {CACHE_VERSION} with the UI?

**Comment by dgsiegel** [comment]
Go to Settings -> Secrets and there you'll find the variable

**Comment by htruong** [comment]
I just changed the `CACHE_KEY` to 0000001 and rebuild. Still, the build log says:

```
Cache restored from key: raspberrypi0-buildroot-2020.08.2-
```

**Comment by dgsiegel** [comment]
where did you test? the last build was 10h ago.

**Comment by htruong** [comment]
https://github.com/showmewebcam/showmewebcam/pull/42/checks?check_run_id=1444007279

build (raspberrypi0)
succeeded 3 hours ago in 4m 45s

**Comment by dgsiegel** [comment]
that one made use of the cache?

**Comment by htruong** [comment]
> that one made use of the cache?

Wouldn't the "cache key" determines the cache or I'm mistaken? If the cache key determines the cache, then it should have said:

```
Cache restored from key: raspberrypi0-buildroot-2020.08.2-0000001
```

should it?

**Comment by dgsiegel** [comment]
yes. or at least ***, as secrets are not printed in the logs. i couldn't reproduce this in my tests, but i agree that it wasn't a 100% deterministic...

**Comment by htruong** [comment]
LGTM, we'll figure out the cache key later.

**Pull request merged**